packages/reporters/src/LogReporter.ts

```diff
@@ -92,9 +123,11 @@ export class NpmLogReporter implements Reporter {
 
     if (entry?.data?.target) {
       const { packageName, task } = entry.data.target;
+      const pkgColor = getColorForPkg(packageName);
+      const pkgTask = this.options.grouped ? `${pkgColor(packageName)} ${chalk.cyan(task)}` : "";
       const normalizedArgs = normalize(getTaskLogPrefix(packageName ?? "<root>", task), msg);
       prefix = normalizedArgs.prefix;
-      msg = normalizedArgs.message;
+      msg = `${normalizedArgs.message} ${pkgTask}`;
```

Comment:
I don't think you need to do this pkg Task thing at the end - you should just change the color for the prefix :)